lib/internal/modules/cjs/loader.js

```diff
@@ -1195,29 +1195,37 @@ let warnRequireESM = true;
 Module._extensions['.js'] = function(module, filename) {
   if (filename.endsWith('.js')) {
     const pkg = readPackageScope(filename);
+    // Function require shouldn't be used in ES modules.
     if (pkg && pkg.data && pkg.data.type === 'module') {
+      const err = new ERR_REQUIRE_ESM(filename);
       if (warnRequireESM) {
```

Comment:
If we are no longer warning, then it may make sense to always include the message instead of just the first time. I'm not sure, open to suggestions here though.

Reply:
Yeap, on my opinion, that warning looks more like an error message, a very explained one.

By reading the errors patern [here](https://github.com/nodejs/node/blob/master/lib/internal/errors.js#L1125), maybe that warning message it's too long for an error message, that warning message (when no longer emitted), could fit very nicely on ERR_REQUIRE_ESM documentation. On my opinion.

Reply:
I would prefer we include the full information in this common pitfall for new users as opposed to requiring them to read the documentation. I think it's a nicer user experience. There's no Node.js policy against long error messages as far as I'm aware.